cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -1283,8 +1287,19 @@ private String getHiveDatabase(@Nullable String namespace) {
     if (namespace == null) {
       return null;
     }
-    String tablePrefix = cConf.get(Constants.Dataset.TABLE_PREFIX);
-    return namespace.equals(Id.Namespace.DEFAULT.getId()) ? namespace : String.format("%s_%s", tablePrefix, namespace);
+    try {
+      if (namespace.equals(Id.Namespace.DEFAULT.getId())) {
+        return namespace;
+      } else if (!Strings.isNullOrEmpty(
+        namespaceQueryAdmin.get(Id.Namespace.from(namespace)).getConfig().getHiveDatabase())) {
+        return namespaceQueryAdmin.get(Id.Namespace.from(namespace)).getConfig().getHiveDatabase();
```

Comment:
Second call not required, we can simply use the result of the first call, right?